src/components/TodoList.tsx

```diff
@@ -0,0 +1,70 @@
+import { FC, useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { Todo } from '../types/Todo';
+import { TodoItem } from './TodoItem';
+
+type Props = {
+  todos: Todo[],
+  toggleCompletedStatus: (
+    todoIds: number[],
+    data: Pick<Todo, 'completed'>,
+  ) => void;
+  onTodoDelete: (todoIds: number[]) => void;
+  handleTitleChange: (todoId: number, title: string) => void;
+};
+
+enum Status {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
+
+export const TodoList: FC<Props> = ({
+  todos,
+  toggleCompletedStatus,
+  onTodoDelete,
+  handleTitleChange,
+}) => {
+  const location = useLocation();
+  const [todosList, setTodosList] = useState<Todo[]>(todos);
+
+  const filterTodosByPathname = () => {
+    if (location.pathname.endsWith(Status.Active)) {
+      const filteredArray = todos.filter(todo => !todo.completed);
+
+      setTodosList(filteredArray);
+
+      return;
+    }
+
+    if (location.pathname.includes(Status.Completed)) {
+      const filteredArray = todos.filter(todo => todo.completed);
+
+      setTodosList(filteredArray);
+
+      return;
+    }
+
+    setTodosList(todos);
+  };
+
+  useEffect(() => {
+    filterTodosByPathname();
+  }, [location, todos]);
```

Comment:
It's better to wrap filterTodosByPathname function into useCallback and move dependencies to the useCallback array. For useEffect just leave a function in the dependency array
```suggestion
  useEffect(() => {
    filterTodosByPathname();
  }, [filterTodosByPathname]);
```